src/library/cratehighlightdelegate.cpp

```diff
@@ -0,0 +1,32 @@
+#include "library/cratehighlightdelegate.h"
+#include "library/treeitem.h"
+#include "trackinfoobject.h"
+#include "libraryfeature.h"
+#include "cratefeature.h"
+
+CrateHighlightDelegate::CrateHighlightDelegate(QObject* parent)
+    : QStyledItemDelegate(parent) {
+
+}
+
+// This will be called to by Qt before painting the TreeView-Item. Set up styles here
+void CrateHighlightDelegate::initStyleOption(QStyleOptionViewItem* option,
+                     const QModelIndex& index) const {
+    if (!index.isValid()) {
+        return;
+    }
+
+    QStyledItemDelegate::initStyleOption(option, index);
+    QStyleOptionViewItemV4 *optionV4 = qstyleoption_cast<QStyleOptionViewItemV4*>(option);
+
+    TreeItem *item = static_cast<TreeItem*>(index.internalPointer());
```

Comment:
The segfault is because the internal pointers in the sidebar can be ```SidebarModel*``` for items that are children of the root and ```TreeItem*``` for all others.